### ISSUE TITLE
URI for mongoDBStore uses only 'test' db

### ISSUE THREAD

**Issue by sagar20896** [mongodb-js/connect-mongodb-session#50]
```
var store = new MongoDBStore(
      {
        uri: 'mongodb://localhost:27017/sessions',
        collection: 'mySessions'
      });
```

I have the above configuration, it must use 'sessions' db but it uses only 'test' db.

**Comment by sagar20896** [comment]
`var store = new MongoDBStore({uri: 'mongodb://localhost:27017/<db>', databaseName: '<db>', collection: 'sessions'});
`
You gotta give your db name at two places for it to work.

**Comment by furanzujin** [comment]
Same issue here and I don't want to duplicate config parameters.

**Comment by shirishp** [comment]
Having the same issue.

**Comment by iPipelineMK** [comment]
The new mongoDB driver requires specifying the DB name, you can likely remove the DB name from the uri property that you are passing into the store if you want to avoid duplication. I didn't want to make it parse the uri property and handle the new format behind the since in cause other options are ever passed in via the uri property.

**Comment by shirishp** [comment]
@iPipelineMK I don't think that's totally correct. If I understand the documentation correctly, the database option is required only if database name is not specified in the connection string.

I confirmed that the library still works as expected if I delete database option from the configuration but provide it in database URI.
https://github.com/mongodb-js/connect-mongodb-session/blob/master/index.js#L19

This change is breaking a lots of software which relies on infrastructure providers supplying connection URI as environment variable. The library would still work as of today even if we remove database name config option.

IMO, even if we want to make this change, it should be thought through in details.

//cc @vkarpov15

**Comment by shirishp** [comment]
As a workaround for now, I am using this poor man's parsing function (not tested enough to be used in production).

```javascript
export const extractDatabaseName = uri => uri.split('/')[3].split('?')[0];
```

with test
```javascript
  it('should extract database name from the connection URI', () => {
    const connectionURI = 'mongodb://username:password@hostName:13509/database-name?replicaSet=rs-1234';

    const databaseName = extractDatabaseName(connectionURI);
    expect(databaseName).toBe('database-name');
  });
```

**Comment by vkarpov15** [comment]
This only affects 2.0.0, and is because the new 3.0.0 release of the mongodb driver [gives you a client rather than an individual database handle](https://github.com/mongodb/node-mongodb-native/blob/3.0.0/CHANGES_3.0.0.md#api-changes). In hindsight, this change is exceedingly inconvenient, so we'll add a workaround in 2.0.1.

**Comment by vkarpov15** [comment]
As a workaround, specify `databaseName` in the options object.

**Comment by vkarpov15** [comment]
2.0.1 released with the fix :+1:

**Comment by iPipelineMK** [comment]
Thanks!

**Comment by shirishp** [comment]
Thank you! :)

**Comment by mbroadst** [comment]
@vkarpov15 not sure if this helps you or not, but in the v3.0.4 release we added [this feature](https://github.com/mongodb/node-mongodb-native/commit/23b193819f3398fa211733d879adc1e1805a68f7), which allows you to do something like this:

```
MongoClient.connect('mongodb://localhost/mydb', (err, client) => {
  client.db().collection('foo').insertOne()....
});
```
and it will automatically use the db you provided in your connection string. 

Also, as you can see from the commit itself, the database name is not lost but instead stored in an accessible internal variable `client.s.options.dbName`.

HTH!

**Comment by vkarpov15** [comment]
Thanks @mbroadst, that syntax is extremely helpful, I adapted our code to use it in 02fd1a4. Will have to port it to mongoose too so mongoose doesn't have to parse the uri twice

**Comment by Toolito** [comment]
@vkarpov15 It seems that your fix is not published in v2.0.1.
https://github.com/mongodb-js/connect-mongodb-session/commits/2.0.1

Could you please release a new version with this commit ?

**Comment by zbjornson** [comment]
@vkarpov15 the fix in b609830 (v2.0.1) doesn't work for replica sets -- note the bad value for `path`:

```js
> url.parse("mongodb://server-1,server-2/dbname?replicaSet=rs0")
Url {
  protocol: 'mongodb:',
  slashes: true,
  auth: null,
  host: 'server-1',
  port: null,
  hostname: 'server-1',
  hash: null,
  search: '?replicaSet=rs0',
  query: 'replicaSet=rs0',
  pathname: '/,server-2/reactive',
  path: '/,server-2/dbname?replicaSet=rs0',
  href: 'mongodb://server-1/,server-2/dbname?replicaSet=rs0' }
```

Sorry for the noise, but +1 for publishing a new release so that RS works, please!

**Comment by vkarpov15** [comment]
Thanks for the heads up @zbjornson , I wasn't aware that this was broken for replica sets. I just published 2.0.2 with this fix :+1:

**Comment by zbjornson** [comment]
Thank you! :)
